### DISCUSSION TITLE
Plotting Heterogenous Graph

### DISCUSSION THREAD

**Discussion by mprithiv** [pyg-team/pytorch_geometric#3888] [Q&A]
Following the [movielens example](https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html), I have created a heterogeneous graph of structure
`HeteroData(
  y=[100836],
  user={ num_nodes=610 },
  movie={ x=[9742, 404] },
  (user, rates, movie)={
    edge_index=[2, 100836],
    edge_label=[100836, 1]
  }
)`

I am looking to plot the graph using to_networkx function. Seems like the HeteroData cannot be used for plotting the graph. Is there a way to plot the heterogeneous graphs? In my real use case, I would have two edge types

    (buyer, pays, seller) & (buyer, date, seller)

What would be the ideal way to use both edge types when creating an embedding? 

`amount_edge_index, amount_edge_label = load_mapping("processed/simulated_mappings.csv",
                                      src_index_col='buyerId',
                                      src_mapping = user_mapping,
                                      dst_index_col = 'sellerId',
                                      dst_mapping = merchant_mapping,
                                      encoders = {
                                                  'txn_amt': LabelEncoder(dtype=torch.long)
                                      },
                                     
                                     )`

similarly
`date_edge_index, date_edge_label = load_mapping("processed/simulated_mappings.csv",
                                      src_index_col='buyerId',
                                      src_mapping = user_mapping,
                                      dst_index_col = 'sellerId',
                                      dst_mapping = merchant_mapping,
                                      encoders = {'txn_date': SequenceEncoder()
                                      },
                                     
                                     )`

I created a separate edge_label and edge_index. Not sure if this is a correct representation

`data['buyer', ' txn_amt',  'seller'].edge_index = amount_edge_index
data['buyer','txn_date', 'seller'].edge_index = date_edge_index

data['buyer', 'txn_amt', 'seller'].edge_label = amount_edge_label
data['buyer', 'txn_date', 'seller'].edge_label = date_edge_label`

**Comment by rusty1s** [comment]
I think adding support for `HeteroData` visualization would be indeed great, but I guess it's not top prio to integrate that into PyG directly. One alternative is to call `to_networkx` based on the output of `HeteroData.to_homogeneous()`.
